File: algorithms/665. Non-decreasing Array/solution.py
class Solution(object):
    def checkPossibility(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        m = 0
        for i in xrange(1, len(nums)):
            if nums[i] < nums[i-1]:
                if m:
                    return False
                m = 1
                if i == 1 or nums[i-2] <= nums[i]:
                    nums[i-1] = nums[i]
                else:
                    nums[i] = nums[i-1]
        return True

# 325 / 325 test cases passed.
# Status: Accepted
# Runtime: 56 ms

File: algorithms/665. Non-decreasing Array/solution.cc
class Solution {
public:
    bool checkPossibility(vector<int>& nums) {
        int dec = 0;
        for ( int i = 1; dec < 2 && i < nums.size(); ++i ) {
            if ( nums[i] < nums[i-1] ) {
                ++dec;
                if ( i == 1 || nums[i-2] <= nums[i] )
                    nums[i-1] = nums[i];
                else
                    nums[i] = nums[i-1];
            }
        }
        return dec < 2;
    }
};

// 325 / 325 test cases passed.
// Status: Accepted
// Runtime: 33 ms
